pkg/query/api/v1.go

```diff
@@ -123,7 +125,7 @@ func (api *API) Register(r *route.Router) {
 				w.WriteHeader(http.StatusNoContent)
 			}
 		})
-		return prometheus.InstrumentHandler(name, hf)
+		return prometheus.InstrumentHandler(name, gziphandler.GzipHandler(hf))
```

Comment:
Maybe it is smart to use same as Prometheus inside `prometheus/web/api/v1/api.go:148`? 

`httputil.CompressionHandler`

Reply:
For reference: we decided against it after all because of dependency hell.